ocean/src/clans.rs

```diff
@@ -1,39 +1,76 @@
+use std::collections::HashMap;
 
 #[derive(Debug)]
 pub struct ClanSystem {
     // TODO: add necessary fields
+    my_map: HashMap<String, Vec<String>>,
```

Comment:
I think you mentioned this as a comment on my code as well, but I think you're right that a HashSet would be more performant than a Vec. A quick search showed some validation: https://stackoverflow.com/questions/3185226/huge-performance-difference-between-vector-and-hashset